extra/training.py

```diff
@@ -13,6 +13,24 @@ def sparse_categorical_crossentropy(out, Y):
   y = Tensor(y)
   return out.mul(y).mean()
 
+def focal_loss(p : Tensor, y_train : Tensor, 
+               alpha: Tensor = Tensor(0.25, dtype=dtypes.float32), 
+               gamma: Tensor = Tensor(2, dtype=dtypes.float32), 
+               reduction: str = "mean") -> Tensor:
+    p_t = p * y_train + ((Tensor.ones_like(p) - p) * (Tensor.ones_like(y_train) - y_train))
+    ce_loss = -(p_t + 1e-10).log()
+    loss = ce_loss.mul((Tensor.ones_like(p_t) - p_t) ** (gamma))
+    if alpha >= 0:
```

Comment:
if alpha is tensor this will always be true statement

Reply:
Fixed! thanks